### PR TITLE
Music Genre Classification Model

### PR DESCRIPTION
**Pull request by fspzar123**
Closes: #52 

- **Title** : Music Genre Classification Model
- **Name:** Filbert Shawn
- **Idenitfy yourself:** SSOC' 24

### Describe the add-ons or changes you've made 📃
In the field of machine learning, identifying musical genres is challenging because they are complex and diverse. Convolutional Neural Networks (CNN) and Support Vector Machines (SVM) are widely used for music genre classification. In this research paper, we develop XGBoost, SVM, Random Forest and KNN approach for music genre classification using the Kaggle GTZAN dataset. The proposed model achieves an accuracy of 90%, 75%, 81% and 80% on the training and validation sets respectively. 

### Type of change ☑️

What sort of change have you made:

- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### How Has This Been Tested? ⚙️

Tested all models used in the project

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/README.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/AI-Code/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### Screenshots 📷

Screenshot 2024-06-30 095835

### Note to reviewers 📄

Please review and let me know if there are any changes needed

### CODE REVIEW

**Review by Avdhesh-Varshney — COMMENTED** [review]
@fspzar123 don't commit in main branch and don't delete/edit any other files and docs. You have to re-fork the repo and create your own branch, raise the pr.

Attach the public collab link.

image
